packages/application-shell/src/configure-apollo.js

```diff
@@ -1,18 +1,32 @@
 import ApolloClient from 'apollo-client';
+import apolloVersion from 'apollo-client/version';
 import { ApolloLink } from 'apollo-link';
 import apolloLogger from 'apollo-link-logger';
 import { createHttpLink } from 'apollo-link-http';
 import {
   InMemoryCache,
   IntrospectionFragmentMatcher,
 } from 'apollo-cache-inmemory';
+import createHttpUserAgent from '@commercetools/http-user-agent';
 import { errorLink, headerLink, tokenRetryLink } from './apollo-links';
 import { isLoggerEnabled } from './utils/logger';
+import version from './version';
+
+const userAgent = createHttpUserAgent({
+  name: 'apollo-client',
+  version: apolloVersion,
+  libraryName: '@commercetools-frontend/application-shell',
```

Comment:
couldn't we import this from package.json?

Reply:
This way we would never spot a custom app as a `libraryName`, right?

Reply:
Hmm, you mean to put here the app name?

Reply:
Yeah. That's what I wanted to achieve with the metrics too. Maybe a `ApplicationShell.setApplicationName` could do the trick. Then here `ApplicationShell.getApplicationName`. However then we require every custom app to set it up. Where is we could also use `readPkgUp` in case the starter (or any custom app) has a valid `name` in the pkg.json.

Reply:
We need it to define in both sdk and app-shell packages btw...

Reply:
I think we should pass the app name to the env.json, so that the bundled code can read it from `window.app.name`

Reply:
And we can enforce the `env.json` to contain the `name` field, we already validate other fields

https://github.com/commercetools/merchant-center-application-kit/blob/6fb7ec398fbcb723fdc76211066e0595c9a45593/packages/mc-html-template/lib/load-env.js#L15-L21

Reply:
That's a nice idea. Would be another breaking release for app-kit I suppose? We could also default it to `unknown`.

Reply:
See https://github.com/commercetools/merchant-center-application-kit/pull/694/commits/9c6873be2f8f3c7fceda45633ff4f9f8510a31d2

WDYT?

Reply:
I would keep it optional for now, so we can test it a bit. Then we can enforce it and cut a breaking release.